common/src/main/java/io/pravega/example/common/Transaction.java

```diff
@@ -0,0 +1,81 @@
+package io.pravega.example.common;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+
+import java.io.Serializable;
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/*
+ *   This class represents the schema for interacting with the Embeddings database
+ * */
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class Transaction implements Serializable {
```

Comment:
This class is used for two purposes. The first purpose is for serializing the transactions in the Pravega stream and this has the JPEG bytes but no embeddings. The second purpose is for storing the output of calculating the embeddings. Trying to fit both purposes in this class leads to messy and confusing code. You should create a new class called TransactionWithEmbedding that has embeddingValues. The Transaction class should not have embeddingValues.